src/Redux/Delegates.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+
+namespace Redux 
+{
+    /// <summary>
+    ///     Represents a method that dispatches an action.
+    /// </summary>
+    /// <param name="action">
+    ///     The action to dispatch.
+    /// </param>
+    /// <returns>
+    ///     Varies
+    /// </returns>
+    public delegate dynamic Dispatcher(object action);
```

Comment:
dynamic let you bypass compile time checks. If you cast an instance as dynamic, you will lose the type safety. https://msdn.microsoft.com/en-us/library/dd264736.aspx

If it's still not clear after the msdn article, feel free to ask me anything on dynamic.